templates/agent/puppet.conf.erb

```diff
@@ -31,7 +31,9 @@
     splaylimit        = <%= scope.lookupvar('::puppet::splaylimit') %>
     runinterval       = <%= scope.lookupvar('::puppet::runinterval') %>
     noop              = <%= scope.lookupvar('::puppet::agent_noop') %>
-    configtimeout     = <%= scope.lookupvar('::puppet::configtimeout') %>
+<% if @configtimeout and !@configtimeout.empty? -%>
```

Comment:
I think use of `@configtimeout` rather than a scope.lookupvar's causing the test failure, since configtimeout may not be in the template scope.